### PR TITLE
Font Resource Generation

### PR DESCRIPTION
**Pull request by carlossless**
Added struct generation for ttf/otf fonts. Also made the String struct to not appear in the generated file, if it contains no actual strings. Hope you don't mind the author addition? I can remove it if so.

### CODE REVIEW

**Comment by carlossless** [comment]
Ah! Forgot tests!

**Comment by carlossless** [comment]
Done!

**Comment by carlossless** [comment]
Added @tonyxiao's filter commit. And realised I didn't include any exclusion filtering, so refactored some of the code for reusing, and added the same thing to mine.

**Comment by mrackwitz** [comment]
Regarding your latest changes: Nice! Even when I'm a big fan of applying the Boy Scout Rule (as "Always leave the campground cleaner than you found it." ⚜) :+1:, we should get that PR now ready to be merged before it starts to diverge further.

Two remaining things:
* Bumping the version doesn't belong in a feature PR. That's part of the release process and should happen, either directly on master or (in the best case) on a dedicated branch, when releasing the next gem version over RubyGems.
* I see the authors in the gemspec also as major contributors and maintainers of the project, who take responsibility in case of issues, review PRs *and* made significant and lasting contributions to the code base. So for now, I think it's too early to add you. But from what I saw, I thought you could be perhaps interested in taking more responsibility in the project. If you want to do that, your support would be very welcome. Once you reach that point, where we both feel confident that you will stick around the project, I'd be more than happy to add you to the gemspec / you can add yourself. ;) Feel free to contact me via Twitter DM or on my Git author mail, if you prefer to give me feedback on that point 1-to-1.

**Comment by carlossless** [comment]
Agreed to both points. Squashed all commits into a single one.

P.S. sorry to @tonyxiao, his commit got caught in the crossfire, and unfortunately all his credit got squashed also.

**Comment by tonyxiao** [comment]
no big deal at all, happy to see this all getting merged in :)

File: spec/unit/analyzer/resources_aggregate_analyzer_spec.rb [hunk 1]
```diff
@@ -38,6 +41,8 @@ def subject
         .expects(:analyze).returns([])
       XCRes::ResourcesAnalyzer::XCAssetsAnalyzer.any_instance
         .expects(:analyze).returns(xcassets_section)
+      XCRes::ResourcesAnalyzer::FontResourcesAnalyzer.any_instance
+          .expects(:analyze).returns([])
```
Comment (mrackwitz) [inline spec/unit/analyzer/resources_aggregate_analyzer_spec.rb]: Should be only indented with two spaces, like the lines before.

File: spec/unit/analyzer/resources_analyzer/font_resources_analyzer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    XCRes::ResourcesAnalyzer::FontResourcesAnalyzer
+  end
+
+  before do
+    @analyzer = subject.new
+    @analyzer.logger = stub('Logger', :log)
+  end
+
+  describe '#analyze' do
+    # TODO
```
Comment (mrackwitz) [inline spec/unit/analyzer/resources_analyzer/font_resources_analyzer_spec.rb]: I know that the other resource analyzers are not all fully covered and there are still open TODOs, but I'd prefer to not add additional ones. I think adding a positive test for each of both methods on base of the fixture project would be fine here.

Comment (carlossless) [inline spec/unit/analyzer/resources_analyzer/font_resources_analyzer_spec.rb]: Okay, will do that a bit later though.

File: Gemfile.lock [hunk 3]
```diff
@@ -1,7 +1,7 @@
 PATH
   remote: .
   specs:
-    xcres (0.4.4)
+    xcres (0.5)
```
Comment (mrackwitz) [inline Gemfile.lock]: This is left from the version bump. `bundle install; git add Gemfile.lock; git commit --amend` should do the job.

**Comment by carlossless** [comment]
I noticed that I had made some wrong assumptions about the way font files are used in Cocoa/CocoaTouch. The correct way would be to parse the ttf/otf header and retrieve the name (or names) contained in it. These are the names that are referenced when using `fontWithName:`. I'm thinking of using [ttfunk](https://github.com/prawnpdf/ttfunk) for atleast working with ttf fonts. Looks like we won't be merging this pull request too soon.

**Comment by stefreak** [comment]
@carlossless but don't you have to add them to the plist to use them anyway? wouldn't it be much more KISS when you just parse the plist?

**Comment by carlossless** [comment]
@stefreak probably, though the plist contains only the filenames for font's that should be use. The filenames are not the same thing as the actual font name, which you use in `fontWithName:`. So my previous problem remains. 

@mrackwitz would implementing something like [opentype.js's font parser](https://github.com/nodebox/opentype.js/blob/master/src/parse.js) just for parsing fonts in xcres, make sense? I'm not sure If I'm willing to implement a whole separate package, and maintain it.

Also need to find the time to get this done...

**Pull request closed without merging**